prow/spyglass/artifactfetchers.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"errors"
+
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+var (
+	ErrCannotParseSource = errors.New("could not create job source from provided source")
+)
+
+// JobSource gets information about a location storing the results of a single Prow job
+type JobSource interface {
```

Comment:
It is not obvious to me why we need to export the `spyglass.JobSource` interface to other packages and it doesn't look like this is happening. So can this just be `type jobSource interface`? 

Please review all your types and all your methods to determine which of these really need to be exported.

Please only export those which need to be exported and make the remainder lower case.

Also this isn't java, why do we need an interface in the first place?

Reply:
I used an interface because there are potentially other sources that artifacts could come from. I wanted to implement them with some consistent operations instead of accessing straight from the fields of a gcs-based job source struct